OSDOCS-3605: Added oc adm upgrade channel command (#45326)
openshift/openshift-docs: modules/update-upgrading-cli.adoc

version   4.9.23     True        False         158m    Cluster version is 4.9.23
----

> . Set your channel to `stable-4.10`:

Reviewer: probably worth linking from here to [the channel docs][1], because `fast-4.10` or `eus-4.10` would be supported choices as well.

[1]: https://docs.openshift.com/container-platform/4.10/updating/understanding-upgrade-channels-release.html

Author: @wking Done. I have pushed the changes. Please check and let me know if it makes sense.

+
[source,terminal]
----

> $ oc adm upgrade channel stable-4.10

Reviewer 2: We should mention that the command might give a warning because the cluster does not have all the channel information when we run this command first time.

Author: @LalatenduMohanty Can we show this in example output, rather than explicitly mentioning it in a procedure?
From an end-user perspective, users will be more interested to read and execute commands. Also this information is displayed in the output , which makes them aware of the warning.

Reviewer 2: The command does not always return warnings. So adding the warning output will  not be the right thing IMO.

----

. Review the current update channel information and confirm that your channel

> is set to `stable-4.10`:

Reviewer: going from 4.9 to 4.10, the main thing you want is the `-4.10` suffix.  `stable` is a reasonable prefix, but other prefixes would work too.

Author: Done!

Reviewer: This sentence still says "confirm that your channel is set to `stable-4.10`, without allowing for the other supported 4.10 channels.

Author: @wking Instead of removing the "Review the current..." step, shall we rewrite it like this:

> Review the current update channel information:
> $ oc get clusterversion -o json|jq ".items[0].spec"
> 
> Example output
> {
>   "channel": "stable-4.10",
>   "clusterID": "990f7ab8-109b-4c95-8480-2bd1deec55ff"
> }
> 

I feel this step would ensure that admin is doing the right thing. What you say ?

> 4.10.0  quay.io/openshift-release-dev/ocp-release@sha256:9c5f0df8b192a0d7b46cd5f6a4da2289c155fd5302dec7954f8f06c878160b8b

Reviewer: we should update this output to match what a current 4.10 `oc` would show against a 4.10 cluster.  This will include your current channel and any available alternatives.  Then the following `channel` subcommand call can be "If you need to change the channel based on your organization requirements..." or some such (no need to call the `channel` subcommand if you're already in your desired channel).

----

> . Based on your organization requirements, set the channel to `stable-4.10`, `fast-4.10`, or `eus-4.10`. For more information about the upgrade channels, see xref:../updating/understanding-upgrade-channels-release.adoc#understanding-upgrade-channels-releases[Understanding upgrade channels and releases]:

Reviewer 3: Can't xref from modules still, unfortunately.

```suggestion
. Based on your organization requirements, set the channel to `stable-4.10`, `fast-4.10`, or `eus-4.10` by running the following command:
```

Reviewer 3: To get around the xref limitation:
1. Open the `updating/updating-cluster-cli.adoc` assembly.
2. Add a line under `include::modules/update-upgrading-cli.adoc[leveloffset=+1]`
3. Put the following on the line immediately after the `include`:
```
[role="_additional-resources"]
.Additional resources
* xref:../updating/understanding-upgrade-channels-release.adoc#understanding-upgrade-channels-releases[Understanding upgrade channels and releases]
```
It's not ideal, but that is the closest we can get to what you want to do here.

Author: Fixed.

+
[source,terminal]
----
$ oc adm upgrade channel stable-4.10
----
+
[IMPORTANT]
====

> For production clusters, you must subscribe to a `stable-\*`, `eus-*`, or `fast-*` channel.

Reviewer: I don't understand why only one of the `*` is escaped, but it seems to render fine, so :shrug:

![image](https://user-images.githubusercontent.com/209920/169085860-b53fe464-bd53-4211-9635-3cee66fefa74.png)

Author: @wking Fixed.

Reviewer 4: I see extra `\`.
![0519](https://user-images.githubusercontent.com/20534196/169252664-eb797281-74ce-4d5a-81f9-e6ed495452fa.png)

Author: @jiajliu 
i can see the following in .adoc file.
> For production clusters, you must subscribe to a `stable-\*`, `eus-*`, or `fast-*` channel.

If I remove `\`, it won't render in the html. So I'm retaining the format which was already there.

Reviewer 4: ack
